orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/service/JobService.java

```diff
@@ -27,11 +30,22 @@
 @Service
 public class JobService {
 
-  @Autowired JobConfigurationProperties jobConfigurationProperties;
+  private final JobConfigurationProperties jobConfigurationProperties;
+
+  private final List<PreconfiguredJobConfigurationProvider> preconfiguredJobConfigurationProviders;
+
+  JobService(
+      @Autowired JobConfigurationProperties jobConfigurationProperties,
+      @Autowired(required = false)
+          List<PreconfiguredJobConfigurationProvider> preconfiguredJobConfigurationProviders) {
```

Comment:
We could use `Optional` here, it's a pattern we've used before for the `ExtensionPoint` objects.

Reply:
@jonsie : Converted to Object provider.